src/values/traits.rs

```diff
@@ -22,6 +22,30 @@ macro_rules! trait_value_set {
     );
 }
 
+/// Represents an aggregate value, built on top of other values.
+pub trait AggregateValue: AnyValue {
```

Comment:
I think this should be `pub trait AggregateValue: BasicValue`? Since `AggregateValue` is a subset of `BasicValue`?

Reply:
Oh yep, good catch. I'll fix this as soon as I can. No other problem to fix here?

Reply:
I'll give it one final look before I merge, but I did not see anything else